velox/expression/tests/ExprTest.cpp

```diff
@@ -2095,28 +2101,115 @@ TEST_F(ExprTest, peeledConstant) {
   }
 }
 
+namespace {
+template <typename T>
+struct AlwaysThrowsFunction {
+  template <typename TResult, typename TInput>
+  FOLLY_ALWAYS_INLINE void call(TResult&, const TInput&) {
+    VELOX_FAIL();
+  }
+};
+} // namespace
+
 TEST_F(ExprTest, exceptionContext) {
   auto data = makeRowVector({
       makeFlatVector<int32_t>({1, 2, 3}),
       makeFlatVector<int32_t>({1, 2, 3}),
   });
 
+  registerFunction<AlwaysThrowsFunction, int32_t, int32_t>({"always_throws"});
+
+  // Disable saving vector and expression SQL on error.
+  FLAGS_velox_save_input_on_expression_any_failure_path = "";
+  FLAGS_velox_save_input_on_expression_system_failure_path = "";
+
+  try {
+    evaluate("always_throws(c0) + c1", data);
+    FAIL() << "Expected an exception";
+  } catch (const VeloxException& e) {
+    ASSERT_EQ("always_throws(c0)", e.context());
+    ASSERT_EQ("plus(always_throws(c0), c1)", e.topLevelContext());
+  }
+
   try {
     evaluate("c0 + (c0 + c1) % 0", data);
     FAIL() << "Expected an exception";
   } catch (const VeloxException& e) {
     ASSERT_EQ("mod(cast((plus(c0, c1)) as BIGINT), 0:BIGINT)", e.context());
     ASSERT_EQ(
         "plus(cast((c0) as BIGINT), mod(cast((plus(c0, c1)) as BIGINT), 0:BIGINT))",
-        trimInputPath(e.topLevelContext()));
+        e.topLevelContext());
+  }
 
-    auto inputPath = extractInputPath(e.topLevelContext());
-    auto copy = restoreVector(inputPath);
-    assertEqualVectors(data, copy);
+  try {
+    evaluate("c0 + (c1 % 0)", data);
+    FAIL() << "Expected an exception";
+  } catch (const VeloxException& e) {
+    ASSERT_EQ("mod(cast((c1) as BIGINT), 0:BIGINT)", e.context());
+    ASSERT_EQ(
+        "plus(cast((c0) as BIGINT), mod(cast((c1) as BIGINT), 0:BIGINT))",
+        e.topLevelContext());
```

Comment:
Could we verify that the inputs are not saved if there is not configured? Or from the exception message doesn't contain the data and sql path? Thanks!